src/main/java/org/testng/internal/Invoker.java

```diff
@@ -579,6 +580,29 @@ private ITestResult invokeMethod(
       ITestNGMethod[] afterMethods,
       ConfigurationGroupMethods groupMethods,
       FailureContext failureContext) {
+
+    if(System.getProperties().containsKey("testng.newInstancePerMethod") ||
```

Comment:
Can we please have this logic that determines if this feature should be enabled or not, to `org.testng.internal.RuntimeBehavior` (we are currently trying to ensure that all such feature toggle logic sit in one place).

Reply:
On a side note if this feature is driven by an annotation, then why would one need the additional toggles via JVM argument and environment variable? The tests would fail anyway. Since annotations are meta data at compile time, wouldn't the presence of the annotation alone suffice ?

Reply:
The code you suggested is exactly what I'm trying to get away from.  Workaround code can make readability and maintenance harder by unnecessarily complicating things.  This doesn't seem to be the only time this has been asked: https://stackoverflow.com/questions/15632854/how-to-force-testng-create-new-instance-of-test-class-for-each-method-if-run-mod

The reason for the system wide (envrionment variables and system property) as well as the annotation was to give choice.  You could use the annotation on the class as a test author to make sure any test code that depends on this functionality get's it.  But teams such as mine which have thousands of functional tests can use the runtime options to apply it globally.

Reply:
sorry, the workaround I was mentioning was the code sample above.  I commented on it in the wrong thread.  I'm currently working on putting the code in RuntimeArguments.  I like that suggestion, however if you think the annotation is sufficient, then that's what I'll do.